plugins/org.eclipse.papyrus.uml.interaction.model/src/org/eclipse/papyrus/uml/interaction/internal/model/commands/NudgeOnRemovalCommand.java

```diff
@@ -0,0 +1,313 @@
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalInt;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.eclipse.emf.common.command.Command;
+import org.eclipse.emf.common.command.CommandWrapper;
+import org.eclipse.emf.common.command.IdentityCommand;
+import org.eclipse.emf.edit.domain.EditingDomain;
+import org.eclipse.gmf.runtime.notation.View;
+import org.eclipse.papyrus.uml.interaction.internal.model.impl.LogicalModelPlugin;
+import org.eclipse.papyrus.uml.interaction.internal.model.impl.MElementImpl;
+import org.eclipse.papyrus.uml.interaction.model.MElement;
+import org.eclipse.papyrus.uml.interaction.model.MExecution;
+import org.eclipse.papyrus.uml.interaction.model.MInteraction;
+import org.eclipse.papyrus.uml.interaction.model.MLifeline;
+import org.eclipse.papyrus.uml.interaction.model.MMessage;
+import org.eclipse.papyrus.uml.interaction.model.MOccurrence;
+import org.eclipse.papyrus.uml.interaction.model.spi.DiagramHelper;
+import org.eclipse.papyrus.uml.interaction.model.spi.LayoutHelper;
+import org.eclipse.uml2.uml.Element;
+
+/**
+ * This command analyses the current graph and fills the white space created by the deletion of elements.
+ */
+public class NudgeOnRemovalCommand extends CommandWrapper {
+
+	private static final int DEFAULT_VERTICAL_OFFSET = 10;
+
+	private EditingDomain editingDomain;
+
+	private Set<MElement<? extends Element>> mElementsToRemove;
+
+	private MInteraction interaction;
+
+	private Collection<Element> elementsToRemove;
+
+	/**
+	 * Creates a new {@link NudgeCommand}.
+	 * 
+	 * @param editingDomain
+	 *            the {@link EditingDomain}
+	 * @param interaction
+	 *            the {@link MInteraction}
+	 * @param elementsToRemove
+	 *            the elements which will be removed
+	 */
+	public NudgeOnRemovalCommand(EditingDomain editingDomain, MInteraction interaction,
+			Collection<Element> elementsToRemove) {
+
+		this.editingDomain = editingDomain;
+		this.interaction = interaction;
+		this.elementsToRemove = elementsToRemove;
+		this.mElementsToRemove = getRemovedMElements();
+
+		List<Command> nudgeCommands = new ArrayList<>();
+		nudgeCommands.addAll(createVerticalNudgeCommands());
+		nudgeCommands.addAll(createHorizontalNudgeCommands());
+
+		/* create the command */
+		if (nudgeCommands.isEmpty()) {
+			command = IdentityCommand.INSTANCE;
+		} else {
+			command = CompoundModelCommand.compose(editingDomain, nudgeCommands);
+		}
+
+	}
+
+	/**
+	 * Maps the removed {@link Element}s to their respective {@link MElement}. Filters out {@link MOccurrence}
+	 * as they are currently just point values connected to other {@link MElement}s.
+	 */
+	private Set<MElement<? extends Element>> getRemovedMElements() {
+		return elementsToRemove.stream()//
+				.map(e -> interaction.getElement(e))//
```

Comment:
Just curious, why not use method references here?  `interaction::getElement`, `Optional::isPresent`, and `Optional::get` would all work.

Reply:
As we are still on Java 6 for most of our open source projects I am not yet that used to Java 8, so I forget about method refs sometimes ;-) I agree that method references are nicer to read